CONTRIBUTING.md

```diff
@@ -0,0 +1,374 @@
+| backend-plugin    | A new backend plugin. Plugin to the Node.js Express server.                           | `@janus-idp/backstage-plugin-<ID>-backend`                    | `plugins/<ID>-backend`                    |
+| scaffolder-module | An module exporting custom actions for `@backstage/plugin-scaffolder-backend`         | `@janus-idp/backstage-plugin-scaffolder-backend-module-<ID>`  | `plugins/scaffolder-backend-module-<ID>`  |
+| plugin-common     | A new isomorphic common plugin package. Serves as an interface between other plugins. | `@janus-idp/backstage-plugin-<ID>-common`                     | `plugins/<ID>-common`                     |
+| backend-module    | A new backend module                                                                  | `@janus-idp/backstage-plugin-<ID>-backend-module-<MODULE_ID>` | `plugins/<ID>-backend-module-<MODULE_ID>` |
+| web-library       | A new web-library package. **Currently not used**                                     | `@janus-idp/backstage-<ID>`                                   | `packages/<ID>`                           |
+| plugin-node       | A new Node.js library plugin package. **Currently not used**                          | `@janus-idp/backstage-plugin-<ID>-node`                       | `plugins/<ID>-node`                       |
+| plugin-react      | A new web library plugin package. **Currently not used**                              | `@janus-idp/backstage-plugin-<ID>-react`                      | `plugins/<ID>-react`                      |
+
+2. **Plugin id**: This value is used in templating and determines the resulting package name and folder name. Please see the table above and look where the `<ID>` is used. As you can see, you don't have to specify the plugin role within the ID value, the role is added to the package name automatically.
+
+Once you finish bootstrapping your plugin folder, please look up the upstream documentation. For example:
+
+- [The structure of a plugin](https://backstage.io/docs/plugins/structure-of-a-plugin)
+- For frontend plugins, see the general [plugin development guide](https://backstage.io/docs/plugins/plugin-development) and [composability system](https://backstage.io/docs/plugins/composability).
+- For backend plugins, see the backend specifici [plugin development guide](https://backstage.io/docs/plugins/backend-plugin/).
+
+### Develop a new plugin together with a local backstage instance
+
+Backstage's support for standalone plugin development is minimal (especially for backend plugins), therefore we include a minimal test instance within this repository.
+
+1. Install the plugin via `yarn workspace [app|backend] add @janus-idp/<PLUGIN_NAME>@*` (`@*` at the end ensures the package is always linked to the local package in the `plugins` folder)
+2. Run `yarn start:backstage`
```

Comment:
Could be useful to add some things that briefly tripped me up:

1. I missed the `backstage-plugin-` prefix when adding plugins the first time
2. I was doing a lot of annoying `git stash`es and "okay, what do I actually care about in the status output" before I tried to find a better way

````suggestion
1. Install the plugin via `yarn workspace [app|backend] add @janus-idp/<PLUGIN_NAME>@*`
   * Example: `yarn workspace app add "@janus-idp/backstage-plugin-nexus-repository-manager@*"`
   * `@*` at the end ensures the package is always linked to the local package in the `plugins` folder
1. Follow the plugin's install instructions
1. (Optional) Tell git to assume the modified files are unchanged, so that
   status/diff commands are representative of your current work:

   ```shell
   # Example, list any dev app files where you don't want changes tracked
   git update-index --assume-unchanged \
       app-config.yaml \
       packages/app/package.json \
       packages/app/src/components/catalog/EntityPage.tsx
   ```

1. Run `yarn start:backstage`
````

One other small hurdle I had getting started was that I needed to create example entities with appropriate annotations, and point Backstage at them. Those could maybe be checked into the repo?

Reply:
Hi @rmartine-ias, thanks for your suggestions. Sorry for a late response, I was on PTO. I'll add that 🙂 

Btw:
> I missed the backstage-plugin- prefix when adding plugins the first time

Yeah, that is a common thing... I hope this will be solved with https://github.com/janus-idp/backstage-plugins/pull/782 where we'll get custom templates/boilerplate for newly created plugins.